sdks/python/apache_beam/runners/worker/log_handler.py

```diff
@@ -108,5 +113,16 @@ def _write_log_entries(self):
 
   def _read_log_control_messages(self, log_control_iterator):
     # TODO(vikasrk): Handle control messages.
-    for _ in log_control_iterator:
-      pass
+    while True:
+      try:
+        for _ in log_control_iterator:
+          pass
+        # iterator is closed
+        return
+      except Exception as ex:
+        # reset the stream
+        print >> sys.stderr, "Logging client failed: {}... resetting".format(ex)
```

Comment:
This `print` is not python 3 compatible. Please use the print(...) syntax.

Reply:
+1^

Reply:
```
print("Logging client failed: {}... resetting".format(ex), file=sys.stderr)
```
If you want to print the stack trace then syntax could be little different.

Reply:
Fixed